src/main.rs

```diff
@@ -271,20 +269,49 @@ const APP: () = {
                 polarity: hal::spi::Polarity::IdleHigh,
                 phase: hal::spi::Phase::CaptureOnSecondTransition,
             })
-            .communication_mode(hal::spi::CommunicationMode::Receiver)
             .manage_cs()
-            .transfer_size(1)
-            .frame_size(16)
+            .suspend_when_inactive()
             .cs_delay(220e-9);
 
-            let mut spi = dp.SPI2.spi(
+            dma_channels.0.set_peripheral_address(
+                &dp.SPI2.txdr as *const _ as u32,
+                false,
+            );
+            dma_channels
+                .0
+                .set_memory_address(&SPI_START as *const _ as u32, false);
+            dma_channels
+                .0
+                .set_direction(hal::dma::Direction::MemoryToPeripherial);
+            dma_channels.0.set_transfer_length(1);
+            dma_channels.0.cr().modify(|_, w| {
+                w.circ()
+                    .enabled()
+                    .psize()
+                    .bits16()
```

Comment:
Given that bits16() is operating on the psize() field in the register, I think it would make sense to put them on the same line.

Reply:
Style is all automatically handled by `cargo fmt --all` and `rustfmt` - this is the output format, so it's what we're going with.

I agree that it's not the most ergonomic, but it's easier to just use `rustfmt` as the source of "correct"